src/Microsoft.Performance.SDK.Runtime/ObjectExtensions.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Performance.SDK.Runtime
+{
+    /// <summary>
+    ///     Contains static (Shared in Visual Basic) methods for
+    ///     interacting with <see cref="object"/> instances.
+    /// </summary>
+    public static class ObjectExtensions
+    {
+        /// <summary>
+        ///     Attempts to dispose the given <see cref="object"/>.
+        ///     If <paramref name="self"/> is <see cref="IDisposable"/>,
+        ///     then the <see cref="SafeDispose(IDisposable)"/> method 
+        ///     will be called for <paramref name="self"/>. Otherwise, this
+        ///     method does nothing.
+        /// </summary>
+        /// <param name="self">
+        ///     The <see cref="object"/> to dispose.
+        /// </param>
+        public static void TryDispose(this object self)
+        {
+            (self as IDisposable)?.SafeDispose();
+        }
+
+        /// <summary>
+        ///     Disposes the given instance of <see cref="IDisposable"/>
+        ///     and enforces the contract that <see cref="IDisposable.Dispose"/>
+        ///     should not throw by catch any <see cref="Exception"/>s that
+        ///     occur.
```

Comment:
this only works when using this extension - the actual implementation itself should be sure never to throw. this feels like a false sense of security

Reply:
This is meant more to be used in the runtime to avoid having try { } catch { } around every call to dispose to avoid an errant extension from blocking cleanup of other things.